front-end/src/app/shared/interceptors/http-error.interceptor.ts

```diff
@@ -1,22 +1,38 @@
-import { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpStatusCode } from '@angular/common/http';
+import {
+  HttpContextToken,
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+  HttpResponse,
+  HttpStatusCode,
+} from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Store } from '@ngrx/store';
 import { userLoggedOutAction } from 'app/store/login.actions';
-import { catchError, Observable, throwError } from 'rxjs';
+import { catchError, Observable, throwError, of } from 'rxjs';
+
+export const ALLOW_ERROR_CODES = new HttpContextToken<number[]>(() => []);
 
 @Injectable()
 export class HttpErrorInterceptor implements HttpInterceptor {
-  constructor(private store: Store) { }
+  constructor(private store: Store) {}
 
   intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
     return next.handle(request).pipe(
       catchError((error: HttpErrorResponse) => {
+        //pass on error code if allowed
+        if (request.context.get(ALLOW_ERROR_CODES).includes(error.status)) {
+          return of(new HttpResponse({ status: 404 }));
```

Comment:
should this use error.status?

`new HttpResponse({ status: error.status })
`

Reply:
YES!

Reply:
nice catch